src/core/Link/RouterLink/RouterLink.tsx

```diff
@@ -0,0 +1,101 @@
+// import React, { Component, ReactNode } from 'react';
+import React, { ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import { RouterLinkStyles } from './RouterLink.baseStyles';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
+import { baseClassName } from '../BaseLink/BaseLink';
+import classnames from 'classnames';
+import { HtmlA } from '../../../reset';
+
+//
+// Source: https://www.benmvp.com/blog/polymorphic-react-components-typescript/
+//
+// Source: https://github.com/emotion-js/emotion/blob/master/packages/styled-base/types/helper.d.ts
+// A more precise version of just React.ComponentPropsWithoutRef on its own
+export type PropsOf<
+  C extends keyof JSX.IntrinsicElements | React.JSXElementConstructor<any>,
+> = JSX.LibraryManagedAttributes<C, React.ComponentPropsWithoutRef<C>>;
+
+type AsProp<C extends React.ElementType> = {
+  /**
+   * An override of the default HTML tag.
+   * Can also be another React component.
+   * @default <a>
+   */
+  asComponent?: C;
```

Comment:
This prop was intended to be named just `as` but I found out it messes up styled-components somehow. When I renamed it to something else like this it started to work perfectly.